### PR TITLE
nds: add BlocksDS toolchain support

### PR DESCRIPTION
**Pull request by asiekierka**
BlocksDS is an alternate DS toolchain created by AntonioND, which I had the opportunity to contribute to and thus naively believe it to be a better/more modern choice. In Angband's case, as the code is already fairly well-optimized, it only really comes down to a ~60KB smaller .nds file (and ~85KB less RAM usage).

I've also added an icon and metadata to the BlocksDS build, as well as updated the ARM7 stub (support the L+R+SELECT+START combination to force-exit in case of a crash, and properly update the RTC).

(The CI failure is due to the Docker container being a little out of date - already notified @AntonioND)

### CODE REVIEW

**Comment by NickMcConnell** [comment]
@timschumi any opinions?

**Comment by AntonioND** [comment]
I've just released version 0.8, and updated the Docker images.

https://github.com/blocksds/sdk/releases/tag/v0.8
https://hub.docker.com/repository/docker/skylyrac/blocksds/tags

File: src/Makefile.blocksds.arm7 [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+CFLAGS		+= -std=gnu11 $(WARNFLAGS) $(DEFINES) $(ARCH) \
+		   -mthumb -mthumb-interwork $(INCLUDEFLAGS) -Os \
+		   -ffunction-sections -fdata-sections \
+		   -fomit-frame-pointer
+
+CXXFLAGS	+= -std=gnu++14 $(WARNFLAGS) $(DEFINES) $(ARCH) \
+		   -mthumb -mthumb-interwork $(INCLUDEFLAGS) -Os \
+		   -ffunction-sections -fdata-sections \
+		   -fno-exceptions -fno-rtti \
+		   -fomit-frame-pointer
```
Comment (timschumi) [inline src/Makefile.blocksds.arm7]: Do we want to put any of these behind an NDS_DEBUG check? The old Makefile excludes at least `-fomit-frame-pointer` when setting that.

`-DNDEBUG` also seems to be missing.

File: src/Makefile.blocksds.arm9 [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+CFLAGS		+= -std=gnu11 $(WARNFLAGS) $(DEFINES) $(ARCH) \
+		   -mthumb -mthumb-interwork $(INCLUDEFLAGS) -Os \
+		   -ffunction-sections -fdata-sections \
+		   -fomit-frame-pointer
+
+CXXFLAGS	+= -std=gnu++14 $(WARNFLAGS) $(DEFINES) $(ARCH) \
+		   -mthumb -mthumb-interwork $(INCLUDEFLAGS) -Os \
+		   -ffunction-sections -fdata-sections \
+		   -fno-exceptions -fno-rtti \
+		   -fomit-frame-pointer
```
Comment (timschumi) [inline src/Makefile.blocksds.arm9]: Same here.

**Review by timschumi — COMMENTED** [review]
Sounds generally reasonable to me, just a few comments.

File: src/Makefile.blocksds.arm7 (hunk 1 above)
Comment (asiekierka) [reply src/Makefile.blocksds.arm7]: Resolved, and updated the file/RAM size comparison to take this into account.

**Pull request merged**